controllers/operator/apigateway_controller_test.go

```diff
@@ -287,35 +302,45 @@ var _ = Describe("API-Gateway Controller", func() {
 			Expect(apiGatewayCR.Status.State).To(Equal(operatorv1alpha1.Warning))
 			Expect(apiGatewayCR.Status.Description).To(Equal("There are APIRule(s) that block the deletion of API-Gateway CR. Please take a look at kyma-system/api-gateway-controller-manager logs to see more information about the warning"))
 			Expect(apiGatewayCR.GetObjectMeta().GetFinalizers()).To(ContainElement(ApiGatewayFinalizer))
+
+			Expect(apiGatewayCR.Status.Conditions).To(ContainElement(MatchFields(IgnoreExtras, Fields{
+				"Type":    Equal(conditions.DeletionBlockedExistingResources.Condition().Type),
+				"Message": ContainSubstring(": APIRule/api-rule-0, APIRule/api-rule-1, APIRule/api-rule-2, APIRule/api-rule-3, APIRule/api-rule-4"),
+				"Status":  Equal(metav1.ConditionFalse),
+			})))
 		})
 
 		It("Should not delete API-Gateway CR if there are any ORY Oathkeeper Rules on cluster", func() {
 			// given
 			now := metav1.NewTime(time.Now())
-			apiGatewayCR := &operatorv1alpha1.APIGateway{ObjectMeta: metav1.ObjectMeta{
-				Name:              apiGatewayCRName,
-				Namespace:         testNamespace,
-				DeletionTimestamp: &now,
-				Finalizers:        []string{ApiGatewayFinalizer},
-			},
-			}
-			oryRule, err := runtime.DefaultUnstructuredConverter.ToUnstructured(&oryv1alpha1.Rule{
+			apiGatewayCR := &operatorv1alpha1.APIGateway{
 				ObjectMeta: metav1.ObjectMeta{
-					Name:      "ory-rule",
-					Namespace: "default",
+					Name:              apiGatewayCRName,
+					Namespace:         testNamespace,
+					DeletionTimestamp: &now,
+					Finalizers:        []string{ApiGatewayFinalizer},
 				},
-			})
-
-			r := &unstructured.Unstructured{Object: oryRule}
-			r.SetGroupVersionKind(schema.GroupVersionKind{
-				Group:   "oathkeeper.ory.sh",
-				Version: "v1alpha1",
-				Kind:    "rule",
-			})
-
-			Expect(err).ToNot(HaveOccurred())
-
-			c := createFakeClient(apiGatewayCR, r)
+			}
+			var rules []client.Object
+			for i := 0; i < 6; i++ {
+				// we initialize more than 5 objects, so we validate if we show only 5 in a condition
```

Comment:
I think we are missing the validation, that we show only 5. As far as I can see we only validate that the message contains 5.